contracts/CreditGuild.sol

```diff
@@ -0,0 +1,182 @@
+    event CheckIsMember(bool result);
+    event Stake(uint256 daoBalance);
+
+    constructor(
+      uint256 _membershipFee,
+      uint256 _vouchAmout,
+      address _marketRegistry,
+      address _unionToken,
+      address _underlyingToken
+    ) ERC721("UnionNFT", "UNFT") BaseUnionMember(_marketRegistry, _unionToken, _underlyingToken) {
+        
+        // set membership fee
+        require(_membershipFee >= MIN_MEMBERSHIP_FEE, "membershipFee too low");
+        membershipFee = _membershipFee;
+        vouchAmount = _vouchAmout;
+        isInitialized = false;
+        
+    }
+    // nouns dao v1 logic, nouns dao immutable, ercenumerable
+
+    // Seed the DAO with 3 members
+    function initialize(
+        address member1, 
+        address member2, 
+        address member3
+    ) public onlyOwner {
+        
+        require(!isInitialized, "initialized");
+        isInitialized = true;
+
+        // make sure all addresses are unique
+        require(member1 != member2, "!unique");
+        require(member2 != member3, "!unique");
+        require(member1 != member3, "!unique");
+
+        address[] memory initialMembers = new address[](3);
+        initialMembers[0] = member1;
+        initialMembers[1] = member2;
+        initialMembers[2] = member3;
+
+        require(userManager.checkIsMember(address(this)), "DAO !member");
+
+        for (uint256 i = 0; i < 3; i++) {
+            address member = initialMembers[i];
+
+            // check that this member is vouching for the DAO
+            // TODO: shouldn't this be: getVouchingAmount(address(this), member) ?
+            require(userManager.getVouchingAmount(member, address(this)) > 0, "!vouching");
+
+            // set vouch_amount
+            userManager.updateTrust(member, vouchAmount);
+
+            // mint NFT which is their membership
+            mintNFT(member);
+        } 
+
+        // call stake on behalf of the dao, call register member on behalf of the dao
+
+        emit Initialize(initialMembers);
+
+    }
+
+    function setVouchAmount(uint256 _vouchAmount) public virtual onlyOwner {
+
+        vouchAmount = _vouchAmount;
+        emit SetVouchAmount(vouchAmount);
+
+    }
+
+    function setMembershipFee(uint256 _membershipFee) public virtual onlyOwner {
+
+        membershipFee = _membershipFee;
+        emit SetMembershipFee(membershipFee);
+
+    }
+
+    function burnMembership(address member) public virtual onlyOwner {
+        require(checkIsMember(member), "!member");
+        uint256 membershipId = tokenOfOwnerByIndex(member, 0); 
+        require(ownerOf(membershipId) == member, "!member");
+        _burn(membershipId); 
+        emit BurnMembership(member, membershipId); 
+        
+    }
+
+    function checkIsMember(address potentialMember) public virtual returns(bool) {
+
+        bool result = balanceOf(potentialMember) > 0;
+        emit CheckIsMember(result);
+        return(result);
```

Comment:
Syntax wise this is more common `return result` You can look at using this formatter to provide consistent formatting to your code: https://github.com/prettier-solidity/prettier-plugin-solidity